src/core/Akka.Discovery/Aggregate/AggregateServiceDiscovery.cs

```diff
@@ -0,0 +1,92 @@
+//-----------------------------------------------------------------------
+// <copyright file="Lease.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2020 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2020 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Annotations;
+using Akka.Event;
+using Akka.Util.Internal;
+
+namespace Akka.Discovery.Aggregate
+{
+    [InternalApi]
+    public sealed class AggregateServiceDiscoverySettings
+    {
+        public List<string> DiscoveryMethods { get; }
+
+        public AggregateServiceDiscoverySettings(Configuration.Config config)
+        {
+            var discoveryMethods = config.GetStringList("discovery-methods");
+            if (!discoveryMethods.Any())
+            {
+                throw new ArgumentOutOfRangeException(nameof(discoveryMethods), "At least one discovery method should be specified");
+            }
+
+            DiscoveryMethods = discoveryMethods.ToList();
+        }
+    }
+
+    [InternalApi]
+    public class AggregateServiceDiscovery : ServiceDiscovery
+    {
+        private readonly List<(string, ServiceDiscovery)> _methods;
+        private readonly ILoggingAdapter log;
+
+        public AggregateServiceDiscovery(ExtendedActorSystem system)
+        {
+            var settings = new AggregateServiceDiscoverySettings(system.Settings.Config.GetConfig("akka.discovery.aggregate"));
+            var serviceDiscovery = Discovery.Get(system);
+            _methods = settings.DiscoveryMethods.Select(method => (method, serviceDiscovery.LoadServiceDiscovery(method))).ToList();
+
+            log = Logging.GetLogger(system, nameof(AggregateServiceDiscovery));
+        }
+
+        /// <summary>
+        /// Each discovery method is given the resolveTimeout rather than reducing it each time between methods.
+        /// </summary>
+        public override Task<Resolved> Lookup(Lookup query, TimeSpan resolveTimeout) =>
+            Resolve(_methods, query, resolveTimeout);
+
+        private async Task<Resolved> Resolve(IReadOnlyCollection<(string, ServiceDiscovery)> sds, Lookup query, TimeSpan resolveTimeout)
```

Comment:
`Resolve` is recursive. I think its fine, but it's open for improvents.

Reply:
if someone runs into an issue with it, they we can always change it later